vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java

```diff
@@ -20,31 +25,55 @@
 @Controller
 public class AddSpaceController {
 
-	@Autowired
-	private ISpaceManager spaceManager;
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private ISpaceFactory spaceFactory;
+
+    @RequestMapping(value = "/staff/space/add", method = RequestMethod.GET)
+    public String showAddSpace(Model model) {
+        model.addAttribute("space", new SpaceForm());
+
+        return "staff/space/add";
+    }
+
+    @RequestMapping(value = "/staff/space/add", method = RequestMethod.POST)
+    public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @RequestParam("file") MultipartFile file,
+            Principal principal) throws IOException {
 
-	@Autowired
-	private ISpaceFactory spaceFactory;
+        ISpace space = spaceFactory.createSpace(spaceForm);
+        byte[] bgImage = null;
+        String filename = null;
+        if (file != null) {
+            bgImage = file.getBytes();
+            filename = file.getOriginalFilename();
+        }
+        spaceManager.storeSpace(space, bgImage, filename);
 
-	@RequestMapping(value = "/staff/space/add", method = RequestMethod.GET)
-	public String showAddSpace(Model model) {
-		model.addAttribute("space", new SpaceForm());
+        return "redirect:/staff/space/list";
+    }
 
-		return "staff/space/add";
-	}
+    @RequestMapping(value = "/staff/space/update/{id}", method = RequestMethod.POST)
+    public RedirectView updateSpace(HttpServletRequest request, @PathVariable("id") String id, Model model,
+            @RequestParam("file") MultipartFile file, Principal principal, RedirectAttributes attributes)
+            throws IOException {
 
-	@RequestMapping(value = "/staff/space/add", method = RequestMethod.POST)
-	public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @RequestParam("file") MultipartFile file, Principal principal) throws IOException {
+        byte[] bgImage = null;
+        String filename = null;
+        if (file.isEmpty() || file.equals(null)) {
+            attributes.addAttribute("alertType", "warning");
+            attributes.addAttribute("showAlert", "true");
+            attributes.addAttribute("message", "File Not Found");
+            return new RedirectView(request.getContextPath() + "/staff/space/{id}");
 
-		ISpace space = spaceFactory.createSpace(spaceForm);
-		byte[] bgImage = null;
-		String filename = null;
-		if (file  != null) {
-			bgImage = file.getBytes();
-			filename = file.getOriginalFilename();
-		}
-		spaceManager.storeSpace(space, bgImage, filename);
+        } else if (file != null) {
+            bgImage = file.getBytes();
+            filename = file.getOriginalFilename();
+        }
+        ISpace currentSpace = spaceManager.getFullyLoadedSpace(id);
+        spaceManager.storeSpace(currentSpace, bgImage, filename);
 
-		return "redirect:/staff/space/list";
-	}
+        return new RedirectView(request.getContextPath() + "/staff/space/{id}");
+    }
```

Comment:
is there a reason you're not returning a string?

Reply:
I thought the RedirectAttributes are being leaked to another controller so used RedirectView but understood that's not the case. Changed it back